pkg/controller/client/client.go

```diff
@@ -71,7 +76,10 @@ func (c *Client) Apply(ctx context.Context, in ctrlCli.Object, opts ...ctrlCli.P
 	unstructured.RemoveNestedField(u.Object, "status")
 
 	err = c.Client.Patch(ctx, u, ctrlCli.Apply, opts...)
-	if err != nil {
+	switch {
```

Comment:
Are these parts basically independent fixes?

Reply:
yep, they are, will move them to another PR

Reply:
@ykaliuta  https://github.com/opendatahub-io/opendatahub-operator/pull/1485

Reply:
will rebase once the other. PR gets merged